### PR TITLE
Add functions to upgrade pixels and pixel ranges to higher resolution.

### PR DESCRIPTION
**Pull request by erykoff**
In nest ordering this is a simple operation, but it's convenient to have it in a library place with error checking.

### CODE REVIEW

File: hpgeom/hpgeom.py [hunk 1]
```diff
@@ -458,3 +460,97 @@ def reorder(map_in, ring_to_nest=True):
...
+    if not nest:
+        _pixels = ring_to_nest(nside, _pixels)
+
+    pixel_ranges = np.vstack((_pixels, _pixels + 1)).T
+    pixel_ranges = upgrade_pixel_ranges(nside, pixel_ranges, nside_upgrade)
+    pixels_upgrade = pixel_ranges_to_pixels(pixel_ranges)
+
+    if not nest:
+        pixels_upgrade = nest_to_ring(nside_upgrade, pixels_upgrade)
+        pixels_upgrade.sort()
```
Comment (beckermr) [inline hpgeom/hpgeom.py]: Why are we sorting here?

**Review by beckermr — CHANGES_REQUESTED** [review]
One question.

File: hpgeom/hpgeom.py (hunk 1 above)
Comment (beckermr) [reply hpgeom/hpgeom.py]: To be clear, the tests need things sorted, but sorts are expensive and we should probably leave the decision to sort to the user.

Comment (erykoff) [reply hpgeom/hpgeom.py]: Okay, I can change that and make a note.  And also nobody should ever be doing these sorts of operations on a ring map anyway.  😄

**Review by beckermr — APPROVED** [review]

**Pull request merged**